plugins/android-boot/fu-android-boot-device.c

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright (C) 2022 Dylan Van Assche <me@dylanvanassche.be>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include <fcntl.h>
+#include <string.h>
+
+#include "fu-android-boot-device.h"
+
+#define ANDROID_BOOT_PROC_CMDLINE     "/proc/cmdline"
+#define ANDROID_BOOT_MAX_CMDLINE_SIZE 4096
+#define ANDROID_BOOT_BOOT_SLOT_ARG    "androidboot.slot_suffix"
+#define ANDROID_BOOT_SERIAL_ARG	      "androidboot.serialno"
+#define ANDROID_BOOT_UNKNOWN_VERSION  "0.0.0"
+#define ANDROID_BOOT_SECTOR_SIZE      512
+
+struct _FuAndroidBootDevice {
+	FuUdevDevice parent_instance;
+	const gchar *label;
+	const gchar *boot_slot;
+	const gchar *uuid;
+	const gchar *version;
```

Comment:
I don't think you have any need to store this here.  Just use `fu_device_set_version()` whenever you write it and `fu_device_get_version` whenever you read it.

Reply:
It is to simply the `UNKNOWN_VERSION` assignment. It is set at init and only if the value is reported through the cmdline, it is assigned.

Reply:
Maybe just `fu_device_set_version(dev, UNKNOWN_VERSION)` at the start then